src/main/java/db/migration/V194__Migrate_instructor_userRoles.java

```diff
@@ -0,0 +1,156 @@
+                    // We need to make one userRole for each workgroup the user is an instructor in
+                    Long roleId = 15L; // New role 'instructor'
+
+                    // Identify the instructorType to use in the new userRole
+                    Long instructorTypeId = null;
+
+                    RolesDTO rolesDTO = userRolesDTO.rolesForWorkgroup.get(workgroupId);
+
+                    // If senate/federation and lecturer are present, ignore lecturer role as it is less precise.
+                    // Cannot be senate and federation simultaneously for a workgroup.
+                    if (rolesDTO.isSenate) {
+                        instructorTypeId = 6L;
+                    } else if (rolesDTO.isFederation) {
+                        instructorTypeId = 4L;
+                    } else if (rolesDTO.isLecturer) {
+                        instructorTypeId = 7L;
+                    }
+
+                    // Ensure UserRole does not already exist
+                    Long existingUserRoleId = null;
+
+                    PreparedStatement psExistingUserRole = connection.prepareStatement(
+                            "SELECT * FROM UserRoles WHERE UserId = ? AND WorkgroupId = ? AND RoleId = ?;"
+                    );
+                    psExistingUserRole.setLong(1, userId);
+                    psExistingUserRole.setLong(2, workgroupId);
+                    psExistingUserRole.setLong(3, roleId);
+                    ResultSet rsExistingUserRole = psExistingUserRole.executeQuery();
+
+                    while(rsExistingUserRole.next()) {
+                        existingUserRoleId = rsExistingUserRole.getLong("Id");
+                    }
+
+                    if (existingUserRoleId != null && existingUserRoleId > 0) { continue; }
+
+                    // Create userRole
+                    PreparedStatement psCreateUserRole = connection.prepareStatement(
+                            "INSERT INTO `UserRoles` (UserId, WorkgroupId, RoleId, InstructorTypeId) " +
+                                    "VALUES (?, ?, ?, ?);"
+                            , Statement.RETURN_GENERATED_KEYS
+                    );
+                    psCreateUserRole.setLong(1, userId);
+                    psCreateUserRole.setLong(2, workgroupId);
+                    psCreateUserRole.setLong(3, roleId);
+                    psCreateUserRole.setLong(4, instructorTypeId);
+                    psCreateUserRole.execute();
+                }
+            }
+
+            // Commit changes
+            connection.commit();
+
+        } catch(SQLException e) {
+            e.printStackTrace();
+            return;
+        } finally {
+            try {
+                if (psUsers != null) {
+                    psUsers.close();
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public class UserRolesDTO {
```

Comment:
Why does this migration use DTOs but others do not?

Reply:
They aren't actually DTOs, will fix the naming and simplify usage.